yt_dlp/extractor/tumblr.py

```diff
@@ -107,116 +237,173 @@ class TumblrIE(InfoExtractor):
         'add_ie': ['Instagram'],
     }]
 
+    _providers = {
+        'instagram': 'Instagram',
+        'vimeo': 'Vimeo',
+        'vine': 'Vine',
+        'youtube': 'Youtube',
+    }
+
     def _real_initialize(self):
+        self._ACCESS_TOKEN = None
+        self.get_access_token()
         self._login()
 
+    def get_access_token(self):
+        try:
+            login_page = self._download_webpage(
+                self._LOGIN_URL, None, 'Downloading login page')
+            self._ACCESS_TOKEN = self._search_regex(
+                r'"API_TOKEN":\s*"(\w+)"', login_page, 'API access token')
+        except ExtractorError:
+            self.report_warning('Failed to get access token; metadata will be missing and some videos may not work')
```

Comment:
use `errnote` and `fatal=False` rather than try-catch

Reply:
This was a bug, actually; I intended to handle errors in both download and search but failed to notice that they threw different exceptions.